### PR TITLE
Add DSpace Space

### PR DESCRIPTION
**Pull request by Hwesta**
Add DSpace Space to the storage service
- POST AIP to DSpace using SWORD2 API
- Parse AIP DublinCore metadata and send with SWORD2
- Split package into two bitstreams: objects and metadata
- Set custom permissions on the metadata bitstream
- Return misc attributes on a Package in metadata
- move_from_storage_service accepts a Package parameter - wants discussion

This branch needs some cleanup to remove experiments that didn't work, and clean up commit history.

### CODE REVIEW

File: storage_service/locations/models/dspace.py [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+# This module, alphabetical
+from . import StorageException
+from common import utils
+from location import Location
+
+LOGGER = logging.getLogger(__name__)
+
+
+class DSpace(models.Model):
+    """ Spaces found in the local filesystem of the storage service."""
```
Comment (jrwdunham) [inline storage_service/locations/models/dspace.py]: Docstring should be changed here, no? This is just copy-pasted from models/local_filesystem.py.

File: storage_service/locations/models/dspace.py [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+# Third party dependencies, alphabetical
+from lxml import etree
+import requests
+import sword2
+import jsonfield
+
+# This project, alphabetical
+
+# This module, alphabetical
+from . import StorageException
```
Comment (jrwdunham) [inline storage_service/locations/models/dspace.py]: Imported but not used.

**Review by jrwdunham — COMMENTED** [review]
Aside from 2 very minor comments regarding dspace.py, 👍 from me.

File: storage_service/locations/models/dspace.py (hunk 2 above)
Comment (Hwesta) [reply storage_service/locations/models/dspace.py]: Fixed.

File: storage_service/locations/models/dspace.py (hunk 1 above)
Comment (Hwesta) [reply storage_service/locations/models/dspace.py]: Good catch, fixed.

**Pull request merged**